content/6_Advanced/Lagrange.mdx

```diff
@@ -153,8 +153,8 @@ int main() {
 	ll lo = 0;
 	ll hi = 1e18;
 	while (lo < hi) {
-		ll mid = (lo + hi) / 2;
-		solve_lambda(mid).second <= k ? hi = mid : lo = mid + 1;
+		ll mid = (lo + hi + 1) / 2;
+		solve_lambda(mid).second >= k ? lo = mid : hi = mid - 1;
```

Comment:
so was it wrong before?

Reply:
yeah i think it was wrong before (somehow ACs due to weak tests or smth). i stress tested and found some cases where the code before WAs. hopefully it's correct now lol